waku/nwaku.go

```diff
@@ -501,11 +503,13 @@ func newWakuNode(ctx context.Context, config *WakuConfig, logger *zap.Logger) (*
 	wg.Add(1)
 	n.wakuCtx = C.cGoWakuNew(cJsonConfig, resp)
 	n.MsgChan = make(chan common.Envelope, MsgChanBufferSize)
+	n.TopicHealthChan = make(chan topicHealth, TopicHealthChanBufferSize)
 	n.logger = logger.Named("nwaku")
 	wg.Wait()
 
 	// Notice that the events for self node are handled by the 'MyEventCallback' method
 	C.cGoWakuSetEventCallback(n.wakuCtx)
+	registerNode(n)
```

Comment:
Registering the node when creating the node instead of when starting, as I noticed that if we registered the node later on we missed some events between the time the node actually starts and the moment the callback is executed and the node is registered